org.eclipse.dawnsci.analysis.dataset/src/org/eclipse/dawnsci/analysis/dataset/MockDynamicLazyDataset.java

```diff
@@ -0,0 +1,132 @@
+package org.eclipse.dawnsci.analysis.dataset;
+
+import java.io.IOException;
+
+import org.eclipse.january.IMonitor;
+import org.eclipse.january.dataset.Dataset;
+import org.eclipse.january.dataset.IDataListener;
+import org.eclipse.january.dataset.IDataset;
+import org.eclipse.january.dataset.IDatasetChangeChecker;
+import org.eclipse.january.dataset.IDynamicDataset;
+import org.eclipse.january.dataset.LazyDynamicDataset;
+import org.eclipse.january.dataset.ShapeUtils;
+import org.eclipse.january.dataset.SliceND;
+import org.eclipse.january.io.ILazyLoader;
+
+public class MockDynamicLazyDataset extends LazyDynamicDataset {
+
+	private static final long serialVersionUID = 1L;
+	private int[][] shapeArrays;
+	private int count = 1;
+	private Dataset parent;
+	public boolean allowIncrement = false;
+
+
+	public MockDynamicLazyDataset(int[][] shapes, Dataset parent) {
+		super("mock", parent.getDType(), parent.getElementsPerItem(), shapes[0], parent.getShape(), null);
+		this.parent = parent;
+		loader = new DynamicLazyLoader();
+		shapeArrays = shapes;
+		this.maxShape = parent.getShape();
+	}
+	
+	
+	public void setAllowIncrement(boolean allow) {
+		allowIncrement = allow;
+	}
+	
+	public Dataset getParent() {
+		return parent;
+	}
+	
+	@Override
+	public IDynamicDataset getDataset() {
+		return this;
+	}
+
+
+	@Override
+	public void setMaxShape(int... maxShape) {
+		// TODO Auto-generated method stub
+
+	}
+
+	@Override
+	public void startUpdateChecker(int milliseconds, IDatasetChangeChecker checker) {
+		// TODO Auto-generated method stub
+
+	}
+
+	@Override
+	public boolean refreshShape() {
+
+		if (!allowIncrement) return true;
+
+		if (count < shapeArrays.length) {
+			int[] s = shapeArrays[count++];
+			size = ShapeUtils.calcLongSize(s);
+			shape = s.clone();
+			oShape = shape;
+			oShape.toString();
+		}
+
+		return true;
+	}
+
+	@Override
+	public void addDataListener(IDataListener l) {
+		// TODO Auto-generated method stub
+
+	}
+
+	@Override
+	public void removeDataListener(IDataListener l) {
+		// TODO Auto-generated method stub
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)